common/src/main/java/org/apache/celeborn/common/util/PushDataHeaderUtils.java

```diff
@@ -0,0 +1,104 @@
+  // | attemptId (4 bytes)                    |
+  // | batchId with checksum flag (4 bytes)   |
+  // | length with checksum length (4 bytes)  |
+  // | checksum (4 bytes)                     |
+  //
+  // Fields description:
+  // - mapId: Unique identifier for the map (4 bytes)
+  // - attemptId: Identifier for the attempt (4 bytes)
+  // - batchId with checksum flag:
+  //   -- checksum flag: 1 bit (indicates if batchId has a checksum)
+  //   -- batchId: 31 bits (always positive when represented as an integer)
+  // - length with checksum length: total length of the data + 4 bytes for checksum
+  // - checksum: Always positive integer (4 bytes)
+  public static final int BATCH_HEADER_SIZE = 5 * 4;
+  public static final int BATCH_HEADER_SIZE_WITHOUT_CHECKSUM = BATCH_HEADER_SIZE - 4;
+  public static final int MAP_ID_OFFSET = Platform.BYTE_ARRAY_OFFSET;
+  public static final int ATTEMPT_ID_OFFSET = Platform.BYTE_ARRAY_OFFSET + 4;
+  public static final int BATCH_ID_OFFSET = Platform.BYTE_ARRAY_OFFSET + 8;
+  public static final int LENGTH_OFFSET = Platform.BYTE_ARRAY_OFFSET + 12;
+  public static final int CHECKSUM_OFFSET = Platform.BYTE_ARRAY_OFFSET + 16;
+  public static final int POSITIVE_MASK = 0x7FFFFFFF;
+  public static final int HIGHEST_1_BIT_FLAG_MASK = 0x80000000;
+
+  public static void buildDataHeader(
+      byte[] data, int mapId, int attemptId, int batchId, int length) {
+    assert data.length >= BATCH_HEADER_SIZE;
+    int batchIdWithChecksumFlag = batchIdWithChecksumFlag(batchId);
+    int lengthWithChecksum = length + 4;
+    Platform.putInt(data, MAP_ID_OFFSET, mapId);
+    Platform.putInt(data, ATTEMPT_ID_OFFSET, attemptId);
+    Platform.putInt(data, BATCH_ID_OFFSET, batchIdWithChecksumFlag);
+    Platform.putInt(data, LENGTH_OFFSET, lengthWithChecksum);
+    Platform.putInt(data, CHECKSUM_OFFSET, computeHeaderChecksum32(data));
+  }
+
+  public static int batchIdWithChecksumFlag(int batchId) {
+    return batchId | HIGHEST_1_BIT_FLAG_MASK;
+  }
+
+  public static int batchIdWithoutChecksumFlag(int batchId) {
+    return batchId & POSITIVE_MASK;
+  }
+
+  public static boolean hasChecksumFlag(byte[] data) {
+    int batchId = Platform.getInt(data, BATCH_ID_OFFSET);
+    return (batchId & HIGHEST_1_BIT_FLAG_MASK) != 0;
+  }
+
+  public static int getMapId(byte[] data) {
+    return Platform.getInt(data, MAP_ID_OFFSET);
+  }
+
+  public static int getAttemptId(byte[] data) {
+    return Platform.getInt(data, ATTEMPT_ID_OFFSET);
+  }
+
+  public static int getBatchId(byte[] data) {
+    return batchIdWithoutChecksumFlag(Platform.getInt(data, BATCH_ID_OFFSET));
+  }
+
+  public static int getLength(byte[] data) {
+    return Platform.getInt(data, LENGTH_OFFSET) - 4;
+  }
+
+  public static int computeHeaderChecksum32(byte[] data) {
+    assert data.length >= BATCH_HEADER_SIZE_WITHOUT_CHECKSUM;
+    CRC32 crc32 = new CRC32();
+    crc32.update(data, 0, BATCH_HEADER_SIZE_WITHOUT_CHECKSUM);
```

Comment:
Although 16-byte calculation for CRC can be trivial, there can be enormous push data structures to handle for a Celeborn Worker. I think it would be better to add a switch to let the users enable or disable this feature. For some users, the never meet any issues about this, they can just disable this feature to save the CPU.

Reply:
I have received some reports from users telling me that Celeborn workers are consuming too much CPU during rush hours. If this feature is on by default, this will surely get things worse.
You can make this config a client-side control config and pass it in the ReserveSlots request.

Reply:
OK, I will add a config this week.

Reply:
@FMX Hi, I have added a config to disable header checksum, PTAL

Reply:
CRC is not the fastest choice, I haven't gone through the whole design, and not sure if the algo can be configured like Spark's impl.
https://github.com/apache/spark/pull/47929

Reply:
@pan3793 Thank you for your advice. However, we will verify the checksum in the worker. Therefore, if we have multiple checksum algorithms to choose from, we should consider adding a tag or field in the pushData/pushMerged request. I would prefer to do that at another PR, WDYT?

Reply:
cc @mridulm, do you have any suggestions here?

Reply:
@jiang13021 Seems we only have one bit to mark data chunk includes checksum or not, if so, I would suggest choosing the most efficient algo, namely CRC32C, which is the fastest one that ships in JDK 9+, and port the Hadoop `PureJavaCrc32C` for compatible with JDK 8